api/storage/elasticsearch/elasticsearch.go

```diff
@@ -501,3 +513,40 @@ func (sm *StorageManager) createIndex(index string) {
 	log.WithField("index", index).Info("index created successfully")
 
 }
+
+// checkCurrentIndexDay is charge of getting the duration time until date change (2020-01-01 23:00:00 -> 2020-01-02 00:00:00).
+// when the day changed, the function call to `setCurrentIndexDay` for create new index with current date
+func (sm *StorageManager) checkCurrentIndexDay(now time.Time) {
+
+	zone, _ := now.Zone()
+	location, err := time.LoadLocation(zone)
+	if err != nil {
+		log.WithError(err).WithField("zone", zone).Warn("zone name not found")
+		location = time.UTC
+	}
+
+	tomorrow := startTomorrowDate(now, location)
+	diff := tomorrow.Sub(now)
+
+	log.WithFields(log.Fields{
+		"now":      now,
+		"tomorrow": tomorrow,
+		"duration": diff,
+	}).Info("change index in")
+
+	// wait until duration end
+	<-time.After(diff)
+	sm.setCurrentIndexDay(now)
+
+}
+
+// setCurrentIndexDay create new index with the given
+func (sm *StorageManager) setCurrentIndexDay(dt time.Time) {
```

Comment:
I would split that function into two, one getter for index name and one setter.
I would take the createIndex logic out of this function as the setCurrentIndexDay doesn't represent it.
What happens if the index already exist?

Reply:
I think that the function should create the index and set the new index in the struct 
But, you absolutely right. i should handle the error when i have a problem to create a new index.
I have fixed the error handling